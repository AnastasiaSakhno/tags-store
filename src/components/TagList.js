import React from 'react'
import PropTypes from 'prop-types'
import Tag from './Tag'

const TagList = ({ linkId, tags, onDestroy }) => {
  const toRender = tags.map( (tag, index) => (
    <Tag name={ tag.name } linkId= { linkId } key={ linkId + index } onDestroy={ onDestroy }/>
  ))

  return (<div className="tag-list">{ toRender }</div>)
}

TagList.propTypes = {
  linkId: PropTypes.string,
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string
    })
  ).isRequired,
  onDestroy: PropTypes.func
}

export default TagList
