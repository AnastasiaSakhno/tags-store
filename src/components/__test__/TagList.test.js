import React from 'react'
import renderer from 'react-test-renderer'
import TagList from '../TagList'
import uniqId from '../../utils/uniq-id'

const linkId = uniqId()
const one = {
  linkId: linkId,
  name: 'one'
}
const two = {
  linkId: linkId,
  name: 'two'
}
const tags = [one, two]
const removeTag = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<TagList linkId={ linkId } tags={ tags } onDestroy={ removeTag }/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
