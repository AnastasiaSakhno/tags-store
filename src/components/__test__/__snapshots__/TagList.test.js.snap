// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tag-list"
>
  <span
    className="tag"
  >
    one
    <span
      className="iconbutton-box"
      onClick={[Function]}
    >
      <i
        className="fa fa-remove "
      />
    </span>
  </span>
  <span
    className="tag"
  >
    two
    <span
      className="iconbutton-box"
      onClick={[Function]}
    >
      <i
        className="fa fa-remove "
      />
    </span>
  </span>
</div>
`;
